stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -148,6 +151,48 @@ internal class ChannelLogic(
             }
     }
 
+    override suspend fun onHideChannelPrecondition(
+        channelType: String,
+        channelId: String,
+        clearHistory: Boolean,
+    ): Result<Unit> {
+        return try {
+            Pair(channelType, channelId).toCid()
+            Result.success(Unit)
+        } catch (e: Exception) {
+            when (e) {
+                is IllegalArgumentException, is IllegalStateException -> Result.error(ChatError("CID is not valid"))
+                else -> throw e
```

Comment:
We shouldn't throw an error here. Maybe instead of `when` we can call always return `ChatError(cause = e)`?